optimade/server/utils.py

```diff
@@ -1,56 +1,15 @@
-import urllib
 import traceback
 
-from typing import Dict, Any, Union, List
-from datetime import datetime
+from typing import Dict, Any
 
 from fastapi.encoders import jsonable_encoder
-from starlette.exceptions import HTTPException as StarletteHTTPException
 from starlette.requests import Request
 from starlette.responses import JSONResponse
 
-from optimade.models import (
-    ToplevelLinks,
-    ResponseMeta,
-    ResponseMetaQuery,
-    Provider,
-    Error,
-    ErrorResponse,
-    EntryResource,
-    StructureResource,
-    ReferenceResource,
-    EntryResponseMany,
-    EntryResponseOne,
-)
+from optimade.models import Error, ErrorResponse
+from optimade.server.routers.utils import meta_values
 
 from .config import CONFIG
-from .entry_collections import EntryCollection
-from .deps import EntryListingQueryParams, SingleEntryQueryParams
-
-
-def meta_values(
-    url: str,
-    data_returned: int,
-    data_available: int,
-    more_data_available: bool,
-    **kwargs,
-) -> ResponseMeta:
-    """Helper to initialize the meta values"""
-    parse_result = urllib.parse.urlparse(url)
-    provider = CONFIG.provider.copy()
-    provider["prefix"] = provider["prefix"][1:-1]  # Remove surrounding `_`
-    return ResponseMeta(
-        query=ResponseMetaQuery(
-            representation=f"{parse_result.path}?{parse_result.query}"
-        ),
-        api_version="v0.10",
-        time_stamp=datetime.utcnow(),
-        data_returned=data_returned,
-        more_data_available=more_data_available,
-        provider=Provider(**provider),
-        data_available=data_available,
-        **kwargs,
-    )
 
 
 def general_exception(
```

Comment:
Any reason this can't be moved to `exception_handlers` (which I guess could renamed `exceptions`)? Doesn't really fit with the other `utils` submodules now. Also I noticed that we have `models.util`, `server.utils` and `server.routers.utils`, please standardize this however you see fit (if `server.utils` is going, probably better to rename to `server.routers.util` as all imports of it will already be in this PR)!

Reply:
> Any reason this can't be moved to `exception_handlers` (which I guess could renamed `exceptions`)?

I indeed call it this in my own implementation. But I think it's a bit misleading, since one would most likely expect Python exceptions to be the content here...

> Doesn't really fit with the other `utils` submodules now. Also I noticed that we have `models.util`, `server.utils` and `server.routers.utils`, please standardize this however you see fit (if `server.utils` is going, probably better to rename to `server.routers.util` as all imports of it will already be in this PR)!

Ah, yeah. It should all be `utils`. And I am indeed trying to move the functions around to the appropriate places.